### DISCUSSION TITLE
UI Overhaul needed!

### DISCUSSION THREAD

**Discussion by Jomyy** [audacity/audacity#2785] [Ideas & Feature Requests]
How could it be that there is absolutely no consistency between ui controls 
the icons are pixelated... sometimes and other times just blurry sometimes native looking ui parts are used and sometimes blurry windows 98 controls are used 
and sometimes there is a gradient on a button and well sometimes there isn't 

or tbh GRADIENTS ON BUTTONS? sorry if i sound like a total butthole but thats just painful to use imo

sometimes we decide to use one font and in the next instance we use another 

TBH i don't care if the ui looks native or not but it SHOULDN'T be BLURRY or PIXELATED 

sorry i'd love to use Audacity but in its current stage, at least on MacOS, it isn't usable for me 
and thats sad because it would be so COOL to have a working and not 20 years old looking Audacity 

and NO the current ui of audacity does not look TIMELESS it looks dated and flawed

ok thats it. sorry. hit me up if i can help to sketch anything out

**Comment by LWinterberg** [comment]
There is absolutely no disagreement here. 

The parts that look native do so because they *are*, it's the primary benefit of using [wxWidgets](https://docs.wxwidgets.org/trunk/page_screenshots.html), the UI library that currently is being used. However, it comes with the drawback that these native controls were designed in the 90s, so they're not flexible enough for what Audacity is currently doing (eg "display dB values alongside the slider"), so a lot of the things need to be custom made, which in turn causes wxWidgets to make trouble, for example because it doesn't really do HiDPI. 

Because of all that, there currently are plans to switch to Qt instead, however doing so will take quite some time as it's nothing short of "refractor all the code and re-write all UI code". So until then, we're looking to improve the UX instead without touching the visuals too much for now, as can be seen in the [Audacity 3.1 release](https://audacityteam.org/3.1.0-video). 

That said: If you want to make a custom theme, you can do so as described here: https://wiki.audacityteam.org/wiki/Theming_HowTo

It won't make anything less blurry, but it does allow you to make some buttons gradient-less. Feel free to share it in the forum when you're done :) https://forum.audacityteam.org/viewforum.php?f=67
